File: 55-jump-game/jump-game.swift
class Solution {
    func canJump(_ nums: [Int]) -> Bool {
        var maxIndex = 0
        for i in 0..<nums.count {
            if i > maxIndex {
                return false
            }
            maxIndex = max(maxIndex, i + nums[i])
            if maxIndex >= nums.count - 1 {
                return true
            }
        }
        return true
    }
}

File: 55-jump-game/jump-game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        maxReach = 0
        for i in range(len(nums)):
            if i>maxReach:
                return False
            maxReach = max(maxReach, nums[i]+i)
            if maxReach >= len(nums)-1:
                return True
        return True
        